build.zig

```diff
@@ -1,18 +1,113 @@
 const std = @import("std");
 
-pub fn build(b: *std.Build) void {
-    var lib = b.addStaticLibrary(.{
-        .name = "tree-sitter",
-        .target = b.standardTargetOptions(.{}),
-        .optimize = b.standardOptimizeOption(.{}),
+pub fn build(b: *std.Build) !void {
+  const target = b.standardTargetOptions(.{});
+  const optimize = b.standardOptimizeOption(.{});
+
+  const wasm = b.option(bool, "enable-wasm", "Enable Wasm support") orelse false;
+  const shared = b.option(bool, "build-shared", "Build shared library") orelse false;
+  const amalgamated = b.option(bool, "amalgamated", "Build using an amalgamated source") orelse false;
+
+  const lib: *std.Build.Step.Compile = if (!shared) b.addStaticLibrary(.{
+    .name = "tree-sitter",
+    .target = target,
+    .optimize = optimize,
+    .link_libc = true,
+  }) else b.addSharedLibrary(.{
+    .name = "tree-sitter",
+    .pic = true,
+    .target = target,
+    .optimize = optimize,
+    .link_libc = true,
+  });
```

Comment:
It's also possible to specify the version, but it requires the patch, which we don't want.